src/crepe/api/LoopTimer.cpp

```diff
@@ -0,0 +1,63 @@
+
+#include "../facade/SDLContext.h"
+#include "../util/log.h"
+
+#include "LoopTimer.h"
+
+using namespace crepe;
+
+LoopTimer::LoopTimer() {
+	dbg_trace();
+}
+
+LoopTimer & LoopTimer::get_instance() {
+    static LoopTimer instance;
+    return instance;
+}
+
+void LoopTimer::start() {
+    last_frame_time = SDLContext::get_instance().get_ticks();
+    elapsed_time = 0;
+    elapsed_fixed_time = 0;
+    delta_time = 0;
+}
+
+void LoopTimer::update() {
+    uint64_t current_frame_time = SDLContext::get_instance().get_ticks();
+    delta_time = (current_frame_time - last_frame_time) / 1000.0;
+
+    if (delta_time > maximum_delta_time) {
+        delta_time = maximum_delta_time;
+    }
+    delta_time = game_scale;
+    elapsed_time += delta_time;
+    last_frame_time = current_frame_time;
+}
+
+double LoopTimer::get_delta_time() const { return delta_time; }
+int LoopTimer::get_current_time() const { return elapsed_time; }
+
+void LoopTimer::advance_fixed_update() { elapsed_fixed_time += fixed_delta_time; }
+
+double LoopTimer::get_fixed_delta_time() const { return fixed_delta_time; }
+
+void LoopTimer::set_fps(int fps) {
+    this->fps = fps;
+    frame_target_time = 1.0 / fps;
```

Comment:
Same here waarom delen door 1

Reply:
De omgekeerde berekenen is niet een magisch getal